_data-prepper/common-use-cases/s3-logs.md

```diff
@@ -0,0 +1,176 @@
+
+To use an SQS dead-letter queue, perform the following steps:
+
+1. Create a new SQS standard queue to act as your DLQ.
+2. Configure your SQS's redrive policy [to use your DLQ](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-configure-dead-letter-queue.html). Consider using a low value such as 2 or 3 for the "Maximum Receives" setting.
+3. Configure the Data Prepper `s3` source to use `retain_messages` for `on_error`. This is the default behavior.
+
+## Pipeline design
+
+Create a pipeline to read logs from S3, starting with an [`s3`]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/configuration/sources/s3/) source plugin. Use the following example for guidance. 
+
+```yaml
+s3-log-pipeline:
+   source:
+     s3:
+       notification_type: sqs
+       compression: gzip
+       codec:
+         newline:
+       sqs:
+         # Change this value to your SQS Queue URL
+         queue_url: "arn:aws:sqs:<YOUR-REGION>:<123456789012>:<YOUR-SQS-QUEUE>"
+         visibility_timeout: "2m"
+```
+
+Configure the following options according to your use case:
+
+* `queue_url`: This the SQS queue URL and is always unique to your pipeline.
+* `codec`: The codec determines how to parse the incoming data.
+* `visibility_timeout`: Configure this value to be large enough for Data Prepper to process 10 S3 objects. However, if you make this value too large, messages that fail to process will take at least as long as the specified value before Data Prepper retries.
+
+The default values for each option work for the majority of use cases. For all available options for the S3 source, see [`s3`]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/configuration/sources/s3/).
+
+```yaml
+s3-log-pipeline:
+   source:
+     s3:
+       notification_type: sqs
+       compression: gzip
+       codec:
+         newline:
+       sqs:
+         # Change this value to your SQS Queue URL
+         queue_url: "arn:aws:sqs:<YOUR-REGION>:<123456789012>:<YOUR-SQS-QUEUE>"
+         visibility_timeout: "2m"
+       aws:
+         # Specify the correct region
+         region: "<YOUR-REGION>"
+         # This shows using an STS role, but you can also use your system's default permissions.
+         sts_role_arn: "arn:aws:iam::<123456789012>:role/<DATA-PREPPER-ROLE>"
+   processor:
+     # You can configure a grok pattern to enrich your documents in OpenSearch.
+     #- grok:
+     #    match:
+     #      message: [ "%{COMMONAPACHELOG}" ]
+   sink:
+     - opensearch:
+         hosts: [ "https://localhost:9200" ]
+         # Change to your credentials
+         username: "admin"
+         password: "admin"
+         index: s3_logs
+```
+
+## Multiple Data Prepper pipelines
+
+We recommend that you have one SQS queue per Data Prepper pipeline. In addition, you can have multiple nodes in the same cluster reading from the same SQS queue, which doesn't require additional configuration with Data Prepper.
```

Comment:
"We recommend that you create one SQS queue per Data Prepper pipeline. Additionally, multiple nodes in the same cluster can read from the same SQS queue, which doesn't require additional Data Prepper configuration."